### ISSUE TITLE
Is it possible to ignore warnings per query?

### ISSUE THREAD

**Issue by DaveVdE** [dotnet/efcore#30411]
## Ask a question

Is it possible to ignore a warning at the query level? I have a custom query that does the ordering and projection, and I simply can't add an `OrderBy()` here:

```
var candidate = _context.Set<NextFlowResultEntity>().FromSqlRaw(query, parameters.ToArray<object>()).FirstOrDefault();
```
This results in a warning for event `CoreEventId.FirstWithoutOrderByAndFilterWarning`. The query will return multiple results in the correct order, so the warning I get while invoking this is a false positive. I understand that it's impossible for EF to know this, and I know I can ignore the warning globally when configuring the context, but I hoped to be able to just ignore it at this query.

### Include provider and version information

EF Core version: 7.0
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET 6.0
Operating system: Windows 10
IDE: Visual Studio 2022 17.4.4

**Comment by roji** [comment]
@DaveVdE when you compose an operator (e.g. FirstOrDefault above) over FromSqlRaw, EF puts your raw SQL inside a subquery. For example, given the following query:

```c#
_ = ctx.Blogs
    .FromSql($"SELECT TOP(10) * FROM Blogs ORDER BY Blogs.Id")
    .FirstOrDefault();
```

EF generates the following SQL:

```sql
SELECT TOP(1) [b].[Id], [b].[Name]
FROM (
    SELECT TOP(10) * FROM Blogs ORDER BY Blogs.Id
) AS [b]
```

Now, in SQL, the ordering of rows coming out of a subquery isn't guaranteed in any way; in other words, even though the subquery above has an ORDER BY, the outer SELECT isn't guaranteed to return any particular row in the 10 coming out of the subquery. Various factors such as indexing or table size could affect this.

In other words, you really must include an OrderBy in your LINQ query outside the raw SQL, in order for FirstOrDefault to have a deterministic result. If this isn't clear or you think it doesn't apply to your case, please post your actual SQL query, and preferably a minimal, runnable console program.

**Comment by DaveVdE** [comment]
Thank you for the quick response.

Yes, I was not aware of that. Basically we wanted to only have the first result of whatever was produced by our custom query. The query itself is custom mostly because of table lock hints that we apply on it, so I'm certain we can rewrite it so that the actual ordering and projection can be done by EF. 

But this is certainly a "gotcha" that we never considered. Thanks!

**Comment by aomader** [comment]
I am also interested in this feature.

**Comment by leotsarev** [comment]
Same for me
